queries/scm.name.scm

```diff
@@ -0,0 +1,99 @@
+;;!! (aaa) @bbb @ccc
+;;!         ^^^^^^^^
+;;!        xxxxxxxxx
+;;!  ---------------
+(
+  (_
+    _ @dummy
+    .
+    (capture
+      "@" @_.leading
+      name: (identifier) @name.start
+    )
+    (capture)? @name.end
+    .
+  ) @_.domain
+  (#not-type? @_.domain parameters)
+  (#not-type? @dummy capture)
+  (#not-parent-type? @_.domain field_definition)
+  (#insertion-delimiter! @name.start " @")
+)
+
+;;!! eee: (aaa) @bbb @ccc
+;;!              ^^^^^^^^
+;;!             xxxxxxxxx
+;;!  --------------------
+(
+  (field_definition
+    (_
+      _ @dummy
+      .
+      (capture
+        "@" @_.leading
+        name: (identifier) @name.start
+      )
+      (capture)? @name.end
+      .
+    )
+  ) @_.domain
+  (#not-type? @dummy capture)
+  (#insertion-delimiter! @name.start " @")
+)
+
+;;!! (aaa) @bbb @ccc
+;;!         ^^^  ^^^
+;;!        xxxx xxxx
+;;!        ---- ----
+(
+  (_
+    (capture
+      "@" @_.leading
+      name: (identifier) @name
+    ) @_.domain
+  ) @dummy
+  (#not-type? @dummy parameters)
+  (#has-n-children-of-type? @dummy capture 2 -1)
+  (#insertion-delimiter! @name " @")
+)
+
+;;!! (aaa) @bbb @ccc
+;;!        *********
+;;!  --------------- <~ iteration domain
```

Comment:
Given iteration scope is just a scope, I'd be tempted to just use the regular symbols.  I figure this is fine for now because these comments are only semi-formal